### PR TITLE
autotools generated Makefile fix

### PR DESCRIPTION
**Pull request by lordrasmus**
when building rpcapd the .o files are searched in the wrong folder

this patch changes where the .o files are generated

and rpcapd must linked to libcrypt to fix the error  undefined reference to `crypt'

### CODE REVIEW

**Comment by infrastation** [comment]
Could you please explain the problem in more detail and check why the change breaks OpenBSD builds?

**Comment by infrastation** [comment]
(and macOS, for that matter)

**Comment by guyharris** [comment]
> and rpcapd must linked to libcrypt to fix the error undefined reference to `crypt'

Only on platforms that don't have the standard `crypt()` routine in the standard libraries.

macOS, for example, is ***NOT*** one of those platforms:

```
$ cat foo.c
#include <stdio.h>
#include <unistd.h>

int
main(int argc, char **argv)
{
	printf("%s\n", crypt(argv[1], argv[2]));
	return 0;
}
$ cc -o foo foo.c
$ cc -o foo foo.c -lcrypt
ld: library not found for -lcrypt
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

The same thing happens on OpenBSD 6.6:

```
$ cc -o foo foo.c
$ cc -o foo foo.c -lcrypt
ld: error: unable to find library -lcrypt
cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

-lcrypt is, however, necessary on Linux (at least on Ubuntu 20.04), FreeBSD (at least on FreeBSD 12), DragonFly BSD (at least on DragonFly BSD 5.8), and NetBSD (at lest on NetBSD 

This is why we have the code in configure.ac after the line

```
AC_MSG_CHECKING([whether to enable remote packet capture])
```

which checks to see whether -lcrypt is necessary to get `crypt()`, and adds it ***IF*** that's the case.

If that's not working, please

1. indicate on what platform it's not working (OS and version);
2. supply the output of the configure script and the config.log file.

File: rpcapd/Makefile.in [hunk 1]
```diff
@@ -78,7 +78,7 @@ INSTALL_DATA = @INSTALL_DATA@
 # problem if you don't own the file but can write to the directory.
 .c.o:
 	@rm -f $@
-	$(CC) $(FULL_CFLAGS) -c $(srcdir)/$*.c
+	$(CC) $(FULL_CFLAGS) -c $(srcdir)/$*.c -o  $*.o
```
Comment (mcr) [inline rpcapd/Makefile.in:81]: Is $*.o a thing?  I always wanted it to be, but I didn't think it was.

**Comment by guyharris** [comment]
> when building rpcapd the .o files are searched in the wrong folder
> 
> this patch changes where the .o files are generated

Searched in the wrong folder, or generated in the wrong folder?

And, again, on what platform is this a problem?  Presumably it's neither macOS nor OpenBSD, given that your other change is not only not *necessary* on those platforms but *breaks the build* on those platforms (it shouldn't be necessary on any other platforms, given that the configure script checks whether -lcrypt is needed or not).  I tried doing both in-tree and out-of-tree builds with both the tip of the main branch and with the 1.10 release tarball, and they all worked on Ubuntu 20.04.

**Comment by guyharris** [comment]
So, again, on which platforms, and under which circumstances, is compiling rpcapd files *without* a `-o` flag a problem?  And what are the symptoms of the problem?

**Comment by lordrasmus** [comment]
i just came back home and saw the error
im using ubuntu 20.04

now i see that  ./configure --enable-remote enables the build of rpcapd 
and then it works

i run ./configure without --enable-remote 

and then

make -C rpcapd/

make: Verzeichnis „/mnt/m2_2/entwicklung_m2/libpcap/rpcapd“ wird betreten
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./daemon.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./fileconf.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./log.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./rpcapd.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./../rpcap-protocol.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./../sockutils.c
gcc -fvisibility=hidden  -I. -I.. -I. -I./..  -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -DHAVE_CONFIG_H  -g -O2    -c ./../sslutils.c
gcc -fvisibility=hidden -g -O2      \
    -o rpcapd daemon.o fileconf.o log.o rpcapd.o ../rpcap-protocol.o ../sockutils.o ../fmtutils.o ../sslutils.o ../libpcap.a -ldbus-1  -lpthread
gcc: error: ../rpcap-protocol.o: Datei oder Verzeichnis nicht gefunden
gcc: error: ../sockutils.o: Datei oder Verzeichnis nicht gefunden
gcc: error: ../sslutils.o: Datei oder Verzeichnis nicht gefunden
make: *** [Makefile:105: rpcapd] Fehler 1

with --enable-remote rpcap-protocol.c is compiled into to the lippcap
but why is rpcapd linked to libpcap.a and the object ../rpcap-protocol.o is linked to into the binary ?

**Comment by guyharris** [comment]
> i run ./configure without --enable-remote

Which means you don't want remote capture support - no support in libpcap *and no recap daemon*.

This means that then doing

> make -C rpcapd/

is a mistake - you're trying to build something that you told the configure script not to configure to build.

**Comment by guyharris** [comment]
> why is rpcapd linked to libpcap.a and the object ../rpcap-protocol.o is linked to into the binary too ?

Because the functions from rpcap-protocol.o are *not* exported from libpcap.

**Comment by lordrasmus** [comment]
ok, I see the flag -fvisibility=hidden is set

So my change is not necessary

I should have tried run configure with 
--enabled-remote
I thought it just enables a protocol

Maybe I should me mentioned that --enabled-remote enables the build of recapd

**Pull request closed without merging**